packages/core/src/awsService/sagemaker/sshConfig.ts

```diff
@@ -0,0 +1,366 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import { SshConfig } from '../../shared/sshConfig'
+import { Result } from '../../shared/utilities/result'
+import { ToolkitError } from '../../shared/errors'
+import { getLogger } from '../../shared/logger/logger'
+import { getIdeProperties } from '../../shared/extensionUtilities'
+import { showConfirmationMessage } from '../../shared/utilities/messages'
+import { CancellationError } from '../../shared/utilities/timeoutUtils'
+import { getSshConfigPath } from '../../shared/extensions/ssh'
+import { fileExists, readFileAsString } from '../../shared/filesystemUtilities'
+import fs from '../../shared/fs/fs'
+
+const localize = nls.loadMessageBundle()
+
+/**
+ * SageMaker-specific SSH configuration that handles outdated config detection and updates.
+ * Extends the base SshConfig with SageMaker-specific validation logic.
+ */
+export class SageMakerSshConfig extends SshConfig {
+    public override async verifySSHHost(proxyCommand: string) {
+        const configStateResult = await this.readSshConfigState(proxyCommand)
+
+        // If reading config state failed, return the error
+        if (configStateResult.isErr()) {
+            return configStateResult
```

Comment:
This is returning the whole result, intentional? Because comment says we want to return error. 

Not sure how the result is structured but guessing error in a property within.